pyglet/media/player.py

```diff
@@ -478,6 +478,11 @@ def update_texture(self, dt=None):
 
             pyglet.clock.schedule_once(self._video_finished, 0)
             return
+        elif ts > time:
```

Comment:
If this seems too risky (in terms of maybe dropping an occasional frame) due to float arithmetic and timer precision, it could easily be adjusted to `ts > time + frame_duration / 2.` or something more tolerant.